### ISSUE TITLE
build errors about missing WarnF method

### ISSUE THREAD

**Issue by ryanfaerman** [nats-io/nats-streaming-server#736]
I'm attempting to embed the nats-streaming-server in my application. When running `go build` I get the following errors:
```
# github.com/nats-io/nats-streaming-server/server
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/clustering.go:276:28: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to newRaftLog:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/server.go:872:7: cannot use ss.stan.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in assignment:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/server.go:1451:37: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewFileStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/server.go:1454:36: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewSQLStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/server.go:1457:39: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to stores.NewMemoryStore:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
/Users/ryanf/go/pkg/mod/github.com/nats-io/nats-streaming-server@v0.11.2/server/server.go:4421:18: cannot use s.log (type *"github.com/nats-io/nats-streaming-server/logger".StanLogger) as type "github.com/nats-io/nats-streaming-server/logger".Logger in argument to traceSubState:
        *"github.com/nats-io/nats-streaming-server/logger".StanLogger does not implement "github.com/nats-io/nats-streaming-server/logger".Logger (missing Warnf method)
```

I've tried:

- building within my gopath
- building in a module outside my gopath
- clearing the `nats-io` directories in my `$GOPATH/mod` directory
- clearing all references to nats within my `go.mod` and `go.sum` and starting over

Nothing seems to work.

**Comment by kozlovic** [comment]
Not sure about modules, but NATS Streaming server vendors NATS Server code at a version prior to the addition of Warnf for the NATS Server logger. My guess is that when you build, you have locally a gnatsd version that is more recent than the vendored version in NATS Streaming repo.
So if you build your app that depends on NATS Streaming, you probably need to also have the dependency on NATS Server at the same rev than the one NATS Streaming is using, which is in the manifest: https://github.com/nats-io/nats-streaming-server/blob/master/vendor/manifest#L149

**Comment by ryanfaerman** [comment]
That seems to have done the trick. I set my `go.mod` to depend directly on `v1.3.0` and everything compiled. Thanks for your help.

**Comment by im-kulikov** [comment]
I have the same problem..
Go modules ignore nats-streaming-server vendor..
And in go.mod file gnats=1.4.0

**Comment by im-kulikov** [comment]
to fix problem, you need to revert to gnats v1.3.0
